src/parsley/validator.js

```diff
@@ -40,79 +43,96 @@ define('parsley/validator', [
         // everything excluding the opening and closing slashes and the flags
         regexp = regexp.replace(new RegExp('^/(.*?)/' + flags + '$'), '$1');
       }
+
       return new RegExp(regexp, flags);
     }
   };
 
-  var convertArrayRequirement = function(string, length) {
-    var m = string.match(/^\s*\[(.*)\]\s*$/)
-    if (!m)
-      throw 'Requirement is not an array: "' + string + '"';
-    var values = m[1].split(',').map(ParsleyUtils.trimString);
-    if (values.length !== length)
-      throw 'Requirement has ' + values.length + ' values when ' + length + ' are needed';
-    return values;
+  var convertArrayRequirement = function (string, length) {
+    try {
+      var array = $.parseJSON(string);
```

Comment:
This only passed the tests because I mistakenly used `eql` instead of `equal` in `test/validators.js` and that my example happened to not have strings, but I want to parse `[hello, world]` which is not JSON.
I'll restructure this PR so we can pick and choose better.